dli_app/mod_reports/controllers.py

```diff
@@ -117,3 +117,22 @@ def view_report(report_id):
         return redirect(url_for('reports.my_reports'))
 
     return render_template('reports/view.html', report=report)
+
+
+@mod_reports.route('/delete/<int:report_id>', methods=['POST'])
+@login_required
+def delete_report(report_id):
+    """Delete the specified report"""
+    report = Report.query.get(report_id)
+    if report is None:
+        flash(
+            "Report not found!",
+            "alert-warning",
+        )
+    else:
+        db.session.delete(report)
```

Comment:
Recommendation: With this change, anyone can delete ANY report, even if it doesn't belong to him.
Add this check:

elif not current_user.is_admin and not report.user.id == current_user.id:
....flash that user does not have permission

We can handle the issue of reports being "favorited" later.